### PR TITLE
Profile add metrics

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Add some metrics

* `DistributionMode` for has_join_prober and hash_join_builder

Remove some metrics

* `_ignore_rows`
* `_send_request_timer`
* `_wait_response_timer`

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Comment by liuyehcf** [comment]
run starrocks_clang-tidy

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Review by kangkaisen — DISMISSED** [review]

File: be/src/exec/vectorized/hash_joiner.cpp [hunk 1]
```diff
@@ -66,6 +66,8 @@ Status HashJoiner::prepare_builder(RuntimeState* state, RuntimeProfile* runtime_
         runtime_profile->add_info_string("Predicates", _hash_join_node.sql_predicates);
     }
 
+    runtime_profile->add_info_string("DistributionMode", to_string(_hash_join_node.distribution_mode));
+    runtime_profile->add_info_string("JoinOp", to_string(_hash_join_node.join_op));
```
Comment (trueeyu) [inline be/src/exec/vectorized/hash_joiner.cpp:70]: has alread add one JoinType.
```
runtime_profile->add_info_string("JoinType", _get_join_type_str(_join_type));
```

**Comment by trueeyu** [comment]
pay attention to the tscancode:

```
Member variable 'Channel::_current_request_bytes,' is not initialized in the constructor.
```

File: be/src/exec/vectorized/hash_joiner.cpp (hunk 1 above)
Comment (liuyehcf) [reply be/src/exec/vectorized/hash_joiner.cpp:70]: In pipeline-engine, `HashJoinNode::prepare` will not be executed.

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

File: be/src/exec/vectorized/hash_joiner.cpp (hunk 1 above)
Comment (trueeyu) [reply be/src/exec/vectorized/hash_joiner.cpp:70]: runtime_profile->add_info_string("JoinType", _get_join_type_str(_join_type));

should be removed ?

File: be/src/exec/vectorized/hash_joiner.cpp [hunk 2]
```diff
@@ -66,6 +66,8 @@ Status HashJoiner::prepare_builder(RuntimeState* state, RuntimeProfile* runtime_
         runtime_profile->add_info_string("Predicates", _hash_join_node.sql_predicates);
     }
 
+    runtime_profile->add_info_string("DistributionMode", to_string(_hash_join_node.distribution_mode));
```
Comment (trueeyu) [inline be/src/exec/vectorized/hash_joiner.cpp:69]: Preferably a readable string, not 1,2,3

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

File: be/src/exec/vectorized/hash_joiner.cpp (hunk 1 above)
Comment (liuyehcf) [reply be/src/exec/vectorized/hash_joiner.cpp]: I misunderstood, I will remove one of the duplicate code

**Review by ZiheLiu — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**